Modified to use NeuralNetwork in sklearn Pipelines and GridSearchCV

jfs42 opened the pull request:
Primary change is extending from BaseEstimater with ClassifierMixin and RegressorMixin to pick up some default behavior required for sklearn's Pipeline and GridSearchCV.

Used this blog post as a starting point/how to:
http://danielhnyk.cz/creating-your-own-estimator-scikit-learn/

All existing tests are passing, and the NeuralNetworkClassifier has been used in a Pipeline and GridSearchCV configuration with changing parameters.

jef5ez commented on mlrose/__init__.py:
```diff
@@ -9,5 +9,5 @@
                       Knapsack, TravellingSales, Queens, MaxKColor, 
                       CustomFitness)
-from .neural import NeuralNetwork, LinearRegression, LogisticRegression
+from .neural import NeuralNetworkClassifier, NeuralNetworkRegressor, LinearRegression, LogisticRegression
```
Broken because you did not make a NeuralNetworkRegressor.

jfs42 replied:
yup, forgot to take it out; fixed.

jfs42 commented:
Gladly @gkhayes,

I had been having regret at changing the API, I've reverted those changes and rebased from your master to make merging easier.

I am fairly certain that the get_params and set_params are not necessary since they are also implemented by the extended BaseEstimator; but I did not take them out after rebasing.

Regards,
Jason
